File: 1两数之和.swift
class Solution {

    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var hashTable = [Int: Int]()
        
        for i in 0..<nums.count {
            if hashTable.keys.contains(target - nums[i]) && i != hashTable[target-nums[i]]! {
                return [i, hashTable[target-nums[i]]!]
            }
            if !hashTable.keys.contains(nums[i]) {
                hashTable[nums[i]] = i
            }
        }
        return [-1, -1]
    }
}

File: 1两数之和.py
#哈希法
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic = {}
        for index, num in enumerate(nums):
            if num not in dic:
                dic[num] = [index]
            else:
                dic[num].append(index)
            if target - num in dic:
                if num != target - num:
                    return [dic[target-num][0], index]
                else:
                    if len(dic[num]) == 2:
                        return [dic[num][0], dic[num][1]]
                    else:
                        continue
